src/plone/rest/events.py

```diff
@@ -13,6 +13,14 @@ def mark_as_api_request(event):
     """Mark a request with Accept 'application/json' with the IAPIRequest
        interface.
     """
+    # In cors calls there is accept header so we need to force
+
+    if event.request.get('REQUEST_METHOD') == 'OPTIONS':
```

Comment:
Handling all OPTIONS requests as IAPIRequest is a bad idea. It will break WebDAV support. Also HAProxy is doing backend healthchecks with OPTIONS requests. I would propose checking for the "Origin" HTTP header.